### PR TITLE
feat: add automatic inspector metadata keys

### PR DESCRIPTION
**Pull request by il-sairamg**
Automatically set inspector metadata for all elements:
- $componentType: element/component type
- $hasFocus: component focus state (auto-updates)
- $isTransitioning: element transition state (auto-updates)

Only active in dev mode when inspector is enabled.

### CODE REVIEW

File: src/lib/codegenerator/generator.js [hunk 1]
```diff
@@ -170,6 +170,18 @@ const generateElementCode = function (
     renderCode.push(`elementConfigs[${counter}][Symbol.for('isSlot')] = true`)
   }
 
+  // Generate framework inspector data setup (dev mode only)
+  if (isDev === true) {
+    const componentType = templateObject[Symbol.for('componentType')] || 'Element'
```
Comment (michielvandergeest) [inline src/lib/codegenerator/generator.js]: Maybe we should set this for Elements, but only components. That doesn't add too much value and should make this a bit more light-weight

File: src/lib/codegenerator/generator.js [hunk 2]
```diff
@@ -170,6 +170,18 @@ const generateElementCode = function (
...
+    renderCode.push(`
+      if (${elm} !== undefined && typeof ${elm}.setInspectorMetadata === 'function') {
```
Comment (michielvandergeest) [inline src/lib/codegenerator/generator.js]: is the `typeof ${elm}.setInspectorMetadata === 'function'` check really necessary?

File: src/lib/lifecycle.js [hunk 3]
```diff
@@ -82,8 +82,26 @@ export default {
...
       emit(v, this.component[symbols.identifier], this.component)
       // update the built-in hasFocus state variable
-      if (v === 'focus') this.component[symbols.state].hasFocus = true
-      if (v === 'unfocus') this.component[symbols.state].hasFocus = false
+      if (v === 'focus') {
+        this.component[symbols.state].hasFocus = true
+        // Update inspector metadata on component's holder element
+        if (
+          this.component[symbols.holder] &&
+          typeof this.component[symbols.holder].setInspectorMetadata === 'function'
```
Comment (michielvandergeest) [inline src/lib/lifecycle.js:98]: do we need the typeof function check here?

File: src/engines/L3/element.js [hunk 4]
```diff
@@ -564,6 +564,44 @@ const Element = {
...
+  /**
+   * Sets framework-provided inspector metadata
+   * Only sets if inspector is enabled and in dev mode
+   * @param {Object} data - Framework inspector metadata to merge
+   */
+  setInspectorMetadata(data) {
+    // Early return if inspector not enabled (performance optimization)
+    if (inspectorEnabled === false) {
+      return
+    }
```
Comment (michielvandergeest) [inline src/engines/L3/element.js]: double check? since we don't call this function is inspectorEnabled is false to begin with?

File: src/engines/L3/element.js [hunk 5]
```diff
@@ -564,6 +564,44 @@ const Element = {
...
+
+    if (inspectorEnabled === null) {
+      inspectorEnabled = Settings.get('inspector', false)
+      if (inspectorEnabled === false) return
+    }
```
Comment (michielvandergeest) [inline src/engines/L3/element.js]: is this a condition that can happen?

Comment (il-sairamg) [reply src/engines/L3/element.js]: I’ve updated this accordingly so the inspectorEnabled === null case is no longer needed here.

File: src/lib/codegenerator/generator.js (hunk 1 above)
Comment (il-sairamg) [reply src/lib/codegenerator/generator.js]: I've updated this to only set metadata for components (not elements), which makes it lighter-weight

File: src/lib/codegenerator/generator.js (hunk 2 above)
Comment (il-sairamg) [reply src/lib/codegenerator/generator.js]: Yes — it’s a cheap defensive guard to avoid dev-time crashes in edge cases (tests/mocks/destroyed elements).

File: src/engines/L3/element.js (hunk 4 above)
Comment (il-sairamg) [reply src/engines/L3/element.js]: Yes — it’s a safe central guard, so callers can call setInspectorMetadata() without checking inspectorEnabled first.

File: src/lib/lifecycle.js [hunk 6]
```diff
@@ -82,8 +82,26 @@ export default {
       // emit 'public' hook
       emit(v, this.component[symbols.identifier], this.component)
       // update the built-in hasFocus state variable
-      if (v === 'focus') this.component[symbols.state].hasFocus = true
-      if (v === 'unfocus') this.component[symbols.state].hasFocus = false
+      if (v === 'focus') {
+        this.component[symbols.state].hasFocus = true
+        // Update inspector metadata on component's holder element
+        if (
```
Comment (michielvandergeest) [inline src/lib/lifecycle.js:95]: should we do an early exit looking for dev mode? To make sure this is not always executed also in production?

Comment (il-sairamg) [reply src/lib/lifecycle.js:95]: Updated checks for early exit.
